polkadot/cli/src/lib.rs

```diff
@@ -189,6 +190,11 @@ pub fn run<I, T>(args: I) -> error::Result<()> where
 		.into();
 
 	config.database_path = db_path(&base_path).to_string_lossy().into();
+	config.pruning = match matches.value_of("pruning") {
+		Some("archive") => PruningMode::ArchiveAll,
+		None => PruningMode::keep_blocks(256),
+		Some(s) => PruningMode::keep_blocks(s.parse().expect("Invalid pruning mode specified")),
```

Comment:
should return an error rather than panic.
a variant `ErrorKind::Input("msg")` would make sense